packages/app/src/sections/futures/Trade/TransferSmartMarginModal.tsx

```diff
@@ -0,0 +1,271 @@
+import { formatDollars } from '@kwenta/sdk/utils'
+import { wei } from '@synthetixio/wei'
+import dynamic from 'next/dynamic'
+import React, { useCallback, useEffect, useMemo, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import styled from 'styled-components'
+
+import CaretDownIcon from 'assets/svg/app/caret-down-slim.svg'
+import CaretUpIcon from 'assets/svg/app/caret-up-slim.svg'
+import BaseModal from 'components/BaseModal'
+import Button from 'components/Button'
+import { CardHeader } from 'components/Card'
+import Error from 'components/ErrorView'
+import NumericInput from 'components/Input/NumericInput'
+import { FlexDivRowCentered } from 'components/layout/flex'
+import SegmentedControl from 'components/SegmentedControl'
+import Spacer from 'components/Spacer'
+import { selectTransaction } from 'state/app/selectors'
+import { selectSusdBalance } from 'state/balances/selectors'
+import { depositCrossMargin, withdrawCrossMargin } from 'state/futures/actions'
+import {
+	selectAvailableMargin,
+	selectIsSubmittingCrossTransfer,
+	selectPosition,
+	selectWithdrawableMargin,
+} from 'state/futures/selectors'
+import { useAppDispatch, useAppSelector } from 'state/hooks'
+
+type Props = {
+	onDismiss(): void
+	defaultTab: 'deposit' | 'withdraw'
+}
+
+type BalanceStatus = 'low_balance' | 'no_balance' | 'high_balance'
+
+const SocketBridge = dynamic(() => import('../../../components/SocketBridge'), {
+	ssr: false,
+})
+
+const PLACEHOLDER = '$0.00'
+
+const TransferSmartMarginModal: React.FC<Props> = ({ onDismiss, defaultTab }) => {
+	const { t } = useTranslation()
+	const dispatch = useAppDispatch()
+
+	const position = useAppSelector(selectPosition)
+	const submitting = useAppSelector(selectIsSubmittingCrossTransfer)
+	const totalWithdrawable = useAppSelector(selectWithdrawableMargin)
+	const transactionState = useAppSelector(selectTransaction)
+	const susdBalance = useAppSelector(selectSusdBalance)
+	const availableMargin = useAppSelector(selectAvailableMargin)
+
+	const minDeposit = useMemo(() => {
+		const accessibleMargin = position?.accessibleMargin ?? ZERO_WEI
+		const min = MIN_MARGIN_AMOUNT.sub(accessibleMargin)
+		return min.lt(ZERO_WEI) ? ZERO_WEI : min
+	}, [position?.accessibleMargin])
+
+	const [openSocket, setOpenSocket] = useState(false)
+	const [amount, setAmount] = useState('')
+	const [transferType, setTransferType] = useState(defaultTab === 'deposit' ? 0 : 1)
+
+	const susdBal = transferType === 0 ? susdBalance : totalWithdrawable
+
+	const balanceStatus: BalanceStatus = useMemo(
+		() =>
+			availableMargin.gt(ZERO_WEI) || susdBalance.gt(minDeposit)
+				? 'high_balance'
+				: susdBalance.eq(ZERO_WEI)
+				? 'no_balance'
+				: 'low_balance',
+		[availableMargin, minDeposit, susdBalance]
+	)
+
+	useEffect(() => {
+		switch (balanceStatus) {
+			case 'no_balance':
+			case 'low_balance':
+				setOpenSocket(true)
+				break
+			case 'high_balance':
+				setOpenSocket(false)
```

Comment:
I think socket can always be open as it's the only function of deposit here.

Reply:
Sure. Updated.